### ISSUE TITLE
GTK 3.22.25 causes redraw, corruption, and disappearing windows

### ISSUE THREAD

**Issue by lukefromdc** [compiz-reloaded/compiz#90]
Update Nov 2, 2017: fixed this locally by resetting all options in ccsm after installing GTK 3.22.25.

With GTK 3.22.25, commit https://git.gnome.org/browse/gtk+/commit/?h=gtk-3-22&id=72045a1a2defa82692bac043c5123da7944d76ec has caused severe rendering issues.  Problems include intermittant corruption or partial rendering of the desktop image, dialog windows being rendered but not redrawn (thus seeming to be frozen as new rendering overwrites old rendering without clearing it), and some windows being "swallowed up" and disappearing.

Killing and restarting Caja will clear corruption of the desktop image itself and appear to let things seem to mostly work, though the problems soon return, often with the first symptom being a disappearing mate-panel (not set to autohide). Setting Caja not to draw icons on the desktop does NOT fix this.

On switching to another WM or in some cases restarting compiz, windows which had disappeared will then be shown as they should have been, except they have been resized to a fraction of their previous width.

**Comment by lukefromdc** [comment]
Only this version of compiz is affected, version 0.9.13.1 works fine with GTK 3.22.25

**Comment by lukefromdc** [comment]
I just found that unsetting, then resetting my preferences in ccsm fixed this. 

Not sure if I did anything different, but I found that on a test partition predating the GTK 3.22.25 transition, updating GTK3 brought up the problems, and then they did not stop when I rolled back to GTK 3.22.24. That meant some other file had been affected and maybe corrupted. I set up a test user with nothing set in ccsm and compiz worked fine. I then added back plugins and it kept working. Back to my normal user, did the same, and compiz still worked. Back to my main system partition (same home directory though), reinstall compiz-reloaded, no more redraw problems. I'll see if this stays stable, and close this if nobody else has trouble
